### DISCUSSION TITLE
[Bug]: The tasks have been completed but not cleaned up. How to clean up the historical data?

### DISCUSSION THREAD

**Discussion by xinlingqudongX** [taskforcesh/bullmq#2938] [General]
### Version

v5.28.1

### Platform

NodeJS

#### How to clean up old historical data, which has been completed in the record

![image](https://github.com/user-attachments/assets/1d44edfb-4ad6-4f39-9f14-ea6722827f48)

- queue.drain和queue.clean使用过但是无法清理，请问如何清理？
- queue.drain and queue.clean have been used but cannot be cleaned up. How can I clean it up?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by manast** [comment]
You should be able to clean jobs using ```clean```, but if you want to have them automatically cleaned you should use the ```removeOn```settings: https://docs.bullmq.io/guide/workers/auto-removal-of-jobs

**Reply by xinlingqudongX** [reply]
- 我的意思是removeOn没有生效，导致有些数据没有删除，clean和drain方法我已经试过了，也没有清理掉，所以是要手动删除吗
- What I mean is that removeOn did not take effect, resulting in some data not being deleted. I have tried the clean and drain methods, but they did not clean it up. So do I need to delete it manually?

**Reply by roggervalf** [reply]
hey @xinlingqudongX if those jobs are not referenced in any of these states: wait, prioritized, completed, failed, delayed or waiting-children, yes you need to remove them manually. How did you get in that state? We may need to have a way to reproduce it if possible
